Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py

```diff
@@ -51,17 +52,36 @@ def parser(cls, parser, loggers=None):
         )
 
     @classmethod
-    def pickable(cls, commits):
-        filtered_commits = []
+    def pickable(cls, commits, repository, commits_story=None):
+        filtered_in = set()
+        all_commits = dict()
+
+        commits_story = commits_story or CommitsStory()
+
         for commit in commits:
+            commits_story.add(commit)
             title = commit.message.splitlines()[0]
+            all_commits[str(commit)] = commit
             if search('Cherry-pick', title) or search('Versioning', title):
                 continue
-            filtered_commits.append(commit)
+            filtered_in.add(str(commit))
 
         # FIXME: We can eliminate more classes of commits by reasoning about commit relationship
         #    For example, a revert of a non-pickable commit is not itself pickable
-        return filtered_commits
+        for ref in list(filtered_in):
+            commit = all_commits[ref]
+            relationships = Trace.relationships(commit, repository, commits_story=commits_story)
+            if not relationships:
+                continue
+            for rel in relationships:
+                if rel.type in Relationship.PAIRED and str(rel.commit) not in filtered_in:
```

Comment:
Can you explain `:71` thru `:82`?

My understanding: For every commit in `filtered_in`, get its relationships; for every relationship, check to see if any reverts or cherry-picks are not in `filtered_in`, if so, then remove the commit from `filtered_in` and break?

Reply:
I don't understand why we'd want to disqualify a commit because we found a follow up?

Reply:
What about versioning commits? Are we filtering for them here?

Reply:
Does this account for the cherry-picks that will occur to BPP branches?

Reply:
We do filter versioning commits, yes, line 65.

I haven't totally taken into consideration BPP branches yet. BPP branches are only relevant when they're the branches we're scanning, which means we have to consider the branch we're branching from much the way the branch has to consider `main`. That being said, a BPP cherry-pick is already handled, because a BPP cherry-pick will refer to a branch commit which will refer to a `main` commit. The `main` commit shouldn't be in the branch story, therefor the whole chain of commits won't be slotted for cherry-pick.